matchms/filtering/metadata_processing/repair_adduct_and_parent_mass_based_on_smiles.py

```diff
@@ -30,40 +32,37 @@ def repair_adduct_and_parent_mass_based_on_smiles(spectrum_in: Spectrum,
     if spectrum_in is None:
         return None
     changed_spectrum = spectrum_in.clone()
+    smiles_mass = get_monoisotopic_neutral_mass(changed_spectrum.get("smiles"))
+    parent_mass = spectrum_in.get("parent_mass")
 
-    precursor_mz = changed_spectrum.get("precursor_mz")
-    if precursor_mz is None:
-        logger.warning("Precursor_mz is None, first run add_precursor_mz")
-        return changed_spectrum
+    # First check if the given adduct and precursor mz already match the monoisotopic mass of the smiles
+    new_parent_mass = derive_parent_mass_from_precursor_mz(changed_spectrum,
+                                                           estimate_from_adduct=True,
+                                                           estimate_from_charge=False)
+    if new_parent_mass is not None:
+        if abs(new_parent_mass-smiles_mass) < mass_tolerance:
```

Comment:
```suggestion
        if abs(new_parent_mass-smiles_mass) <= mass_tolerance:
```

Reply:
I think `<=` better matches what the docstring says: "Maximum allowed mass difference".